t/autoinstance.t

```diff
@@ -6,24 +6,37 @@ $| = 1;
 
 use lib './lib';
 use t::lib::Test;
-use Test::More tests => 6;
+use Test::More tests => 9;
 use DBR::Config::Scope;
 
-my $dbr = setup_schema_ok( 'sorttest' );
-
-my $dbrh = $dbr->connect( 'newsch' );
-ok(!$dbrh, 'dbr connect does not exist, fails');
+my $dbr = setup_schema_ok( 'sorttest', undef, 'alpha');
+my $schema = DBR::Config::Schema->new(handle => 'sorttest', session => $dbr->session);
+my $instance = $schema->get_instance('master','alpha');
+ok($instance, 'base instance Alpha found');
 
 my $dbinfo = $dbr->connect('dbrconf')->select( -table => 'dbr_instances', -fields => 'instance_id schema_id class dbname username password host dbfile module handle readonly tag' )->[0];
+ok($dbinfo, 'fetch instance (Alpha) data for cloning');
+
+
+## Bravo instance ( testing schema->get_instance )
+
+my $instance = $schema->get_instance( 'master', undef, 'bravo' );
+ok(!$instance, 'instance Bravo has not been created yet');
+
+my $r = $dbr->connect('dbrconf')->insert( -table => 'dbr_instances', -fields => { schema_id => ['d',$dbinfo->{schema_id}], map(($_, $dbinfo->{$_}), qw'class dbfile module'), handle => 'new', tag => 'bravo' } );
+ok($r, 'Created instance Bravo)');
+
+my $instance = $schema->get_instance( 'master', 'bravo' );
+ok($instance, 'instance Bravo retrievable via schema->get_instance');
 
-ok($dbinfo, 'fetch DB config info');
 
-my $r = $dbr->connect('dbrconf')->insert( -table => 'dbr_instances', -fields => { schema_id => ['d',$dbinfo->{schema_id}], map(($_, $dbinfo->{$_}), qw'class dbfile module'), handle => 'newsch' } );
-ok($r, 'save DB config info');
+## Charlie instance ( testing dbr->connect )
 
-$dbrh = $dbr->connect( 'newsch' );
-ok($dbrh, 'dbr connect now exists, succeeds');
+my $r = $dbr->connect('dbrconf')->insert( -table => 'dbr_instances', -fields => { schema_id => ['d',$dbinfo->{schema_id}], map(($_, $dbinfo->{$_}), qw'class dbfile module'), handle => 'new', tag => 'charlie' } );
+ok($r, 'Created instance Charlie');
 
+my $dbrh = $dbr->connect( 'new', undef, 'charlie' );
+ok($dbrh, 'instance Bravo retrievable via dbr->connect');
```

Comment:
Should this be `instance Charlie retrievable` since that's what was in the line above?

Reply:
@ropsmarv This scenario has worked since 654ff0e9 ( 2014-03-23 ) because this code path uses DBR::Config::Instance->lookup() which has an auto-reload behavior when an instance is not found.

I've spiffed it up a bit with instance tagging, but this test guards only against regressions.

The Bravo test is the one that was failing. Now that it is passing, we will no longer have to restart services when adding instances with a new tag.

cc @schellj

Reply:
@dnorman I think @ropsmarv's point was that the instance that was created on line 36 and connected to on line 38 was called `charlie`, but the test label on line 39 says `Bravo` instead of `Charlie`.